src/hypothesis/core.py

```diff
@@ -438,6 +455,44 @@ def process_arguments_to_given(
     return arguments, kwargs, test_runner, search_strategy
 
 
+def create_new_argspec_for_given(original_argspec, generator_kwargs,
+                                 type_hint_strategies):
+    """Make an updated argspec for the wrapped test.
+
+    The new arspec excludes any arguments that were supplied to given,
+    and sets st.infer as the default value for arguments with a type-
+    hint-based strategy, so they can be omitted and the strategy
+    inserted at call time by process_arguments_to_given.
+
+    """
+    new_args = [a for a in original_argspec.args if a not in generator_kwargs]
+    new_kwonlyargs = [a for a in original_argspec.kwonlyargs
+                      if a not in generator_kwargs]
+    annots = {
+        k: v for k, v in original_argspec.annotations.items()
+        if k in (original_argspec.args + original_argspec.kwonlyargs)
+    }
+    annots['return'] = None
+
+    has_strats = [not type_hint_strategies.get(arg, st.nothing()).is_empty
+                  for arg in reversed(new_args)]
+    defaults = (st.infer,) * len(list(itertools.takewhile(bool, has_strats)))
+    if len(defaults) < sum(has_strats):
+        warnings.warn(
```

Comment:
> And just for clarity, this will happen regardless of whether there are any annotations at all?

hmm. Yes, I think that's probably best. It does feel a bit weird, but I think it's still the most consistent thing to do.

Note however that I would strongly encourage you to write the fixtures tests before embarking on this, because if they're going to break with `infer` then they're going to break even harder with this, and it would be useful finding that out first.